SingularityS3Base/src/main/java/com/hubspot/singularity/s3/base/ArtifactManager.java

```diff
@@ -87,6 +97,37 @@ private void downloadAndCheck(RemoteArtifact artifact, Path downloadTo) {
 
     checkFilesize(artifact, downloadTo);
     checkMd5(artifact, downloadTo);
+    checkGpgSignature(artifact, downloadTo);
+  }
+
+  private void checkGpgSignature(RemoteArtifact baseArtifact, Path downloadTo) {
+    if (baseArtifact.getGpgSignatureArtifact().isPresent() && baseArtifact.getGpgSignatureArtifact().get() instanceof RemoteArtifact) {
```

Comment:
Always better to reduce indentation. If you have an if block around an entire method, you should either split the inner contents into its own method or negate the IF and return (my preference.)